### PR TITLE
Add SignalingDrop counts meetingFailed event metric

### PR DESCRIPTION
**Pull request by ldai1**
**Issue #:**

**Description of changes:**
Add SignalingDropped counts meeting event metric. The SignalingDropped is now published as a meeting event in this recent  pr: #1580

**Testing**

1. Have you successfully run `npm run build:release` locally? yes
2. How did you test these changes? 
    1. Copy the code change and paste to CloudWatch dashboard for a preview. 
    2. Open the test app url deployed for testing and join a meeting
    3. Enable Network Link conditioner and select 100% loss. 
    4. The websocket connection will be closed
    5. Check the new signalingDropped metric in cloudwatch 
3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it? N/A
4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? N/A
5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: demos/serverless/template.yaml [hunk 1]
```diff
@@ -545,6 +545,20 @@ Resources:
...
                     }
+                },
+                {
+                    "type": "log",
+                    "x": 0,
+                    "y": 18,
+                    "width": 24,
+                    "height": 6,
+                    "properties": {
+                        "query": "SOURCE \"${Source}\" | filter name in [\"signalingDropped\"]\n| fields fromMillis(@timestamp) as timestamp,\nconcat(attributes.osName, \" \", attributes.osVersion) as operatingSystem,\nconcat(attributes.browserName, \" \", attributes.browserMajorVersion) as browser,\nattributes.meetingStatus as failedStatus,\nattributes.meetingErrorMessage as reason,\nconcat(attributes.signalingOpenDurationMs / 1000, \"s\")  as signalingOpenDurationMs,\nattributes.retryCount as retryCount\n| sort @timestamp desc\n",
```
Comment (simmkyu) [inline demos/serverless/template.yaml:556]:
```
filter name in [\"signalingDropped\"]
```

The SDK does not have the `signalingDropped` event.
https://github.com/aws/amazon-chime-sdk-js/blob/master/src/eventcontroller/EventName.ts

The `signalingDropped` is only available as the meeting history state, so this table won't show any data.
https://aws.github.io/amazon-chime-sdk-js/modules/meetingevents.html#the-meeting-history-attribute

Comment (ldai1) [reply demos/serverless/template.yaml]: Oh I will use the meeting history attribute instead of the meeting event. thanks Kyu!

Comment (simmkyu) [reply demos/serverless/template.yaml]: Yon cannot retrieve the value by accessing: `attributes.meetingHistory.signalingDropped`.

`attributes.meetingHistory` is an array of the history state object.

```
[
  {
    name: "audioInputSelected", 
    timestampMs: 1630302522411
  },
  {
    name: "signalingDropped", 
    timestampMs: 1630302522520
  },
  {
    name: "meetingStartRequested", 
    timestampMs: 1630302525540
  },
  {
    name: "signalingDropped", 
    timestampMs: 1630302525600
  }
]
```

You can only access by `attributes.meetingHistory.indexNumber.name`.

```
attributes.meetingHistory.0.timestampMs | 1630302522411
attributes.meetingHistory.1.name | videoInputSelected
attributes.meetingHistory.1.timestampMs | 1630302522520
attributes.meetingHistory.2.name | meetingStartRequested
attributes.meetingHistory.2.timestampMs | 1630302525540
attributes.meetingHistory.3.name | videoInputUnselected
attributes.meetingHistory.3.timestampMs | 1630302525600
attributes.meetingHistory.4.name | meetingStartFailed
attributes.meetingHistory.4.timestampMs | 1630302526467
```

Once we publish the `signalingDropped` as a meeting event, we can use your previous solution to count or list all `signalingDropped` evens. Let's revisit this PR later.

**Review by simmkyu — CHANGES_REQUESTED** [review]
Yon cannot retrieve the value by accessing: `attributes.meetingHistory.signalingDropped`.

`attributes.meetingHistory` is an array of the history state object like the following: 

```js
[
  {
    name: "audioInputSelected", 
    timestampMs: 1630302522411
  },
  {
    name: "signalingDropped", 
    timestampMs: 1630302522520
  },
  {
    name: "meetingStartRequested", 
    timestampMs: 1630302525540
  },
  {
    name: "signalingDropped", 
    timestampMs: 1630302525600
  }
]
```

In CloudWatch Logs Insights, you can access each element by `attributes.meetingHistory.indexNumber.name`.

```
attributes.meetingHistory.0.timestampMs | 1630302522411
attributes.meetingHistory.1.name | videoInputSelected
attributes.meetingHistory.1.timestampMs | 1630302522520
attributes.meetingHistory.2.name | meetingStartRequested
attributes.meetingHistory.2.timestampMs | 1630302525540
attributes.meetingHistory.3.name | videoInputUnselected
attributes.meetingHistory.3.timestampMs | 1630302525600
attributes.meetingHistory.4.name | meetingStartFailed
attributes.meetingHistory.4.timestampMs | 1630302526467
```

⠀

Once we publish the `signalingDropped` as a meeting event, we can use your previous solution to count or list all `signalingDropped` evens. Let's revisit this PR later.

**Comment by ldai1** [comment]
Thanks, I will pause the pr and revisit it once we publish the `signalingDropped` meeting event.

File: demos/serverless/template.yaml [hunk 2]
```diff
@@ -525,13 +525,27 @@ Resources:
...
                     }
                 },
+                {
+                    "type": "log",
+                    "x": 0,
+                    "y": 36,
+                    "width": 24,
+                    "height": 6,
+                    "properties": {
+                        "query": "SOURCE \"${Source}\" | filter name in [\"signalingDropped\"]\n| fields fromMillis(@timestamp) as timestamp,\nconcat(attributes.osName, \" \", attributes.osVersion) as operatingSystem,\nconcat(attributes.browserName, \" \", attributes.browserMajorVersion) as browser,\nattributes.meetingStatus as failedStatus,\nattributes.meetingErrorMessage as reason,\nconcat(attributes.signalingOpenDurationMs / 1000, \"s\")  as signalingOpenDurationMs,\nattributes.retryCount as retryCount\n| sort @timestamp desc\n",
```
Comment (simmkyu) [inline demos/serverless/template.yaml:542]: `attributes.meetingStatus` and `attributes.meetingErrorMessage` do not exist for the `signalingDropped` event.
https://aws.github.io/amazon-chime-sdk-js/modules/meetingevents.html#meeting-attributes

File: demos/serverless/template.yaml [hunk 3]
```diff
@@ -525,13 +525,27 @@ Resources:
...
+                        "region": "us-east-1",
+                        "stacked": false,
+                        "title": "Siggnaling Dropped count",
```
Comment (simmkyu) [inline demos/serverless/template.yaml:545]: Typo. How about `WebSocket signaling failures`?

**Review by simmkyu — CHANGES_REQUESTED** [review]
Try publishing several `signalingDropped` events for testing and ensure that you see those events correctly on the dashboard. You may use the Network Link Conditioner on macOS to simulate poor internet connectivity to trigger the `signalingDropped` from the Chime SDK serverless demo.

**Review by harshitOberoi — APPROVED** [review]
Apart from Kyu comments everything looks good.

File: demos/serverless/template.yaml (hunk 2 above)
Comment (ldai1) [reply demos/serverless/template.yaml]: Fixed.

File: demos/serverless/template.yaml (hunk 3 above)
Comment (ldai1) [reply demos/serverless/template.yaml]: Good catch. Sounds good to me.

**Comment by ldai1** [comment]
Verified that the signalingClient events can be successfully populated into the new metrics.

**Review by simmkyu — APPROVED** [review]

**Pull request merged**